### PR TITLE
Issue #13672: Kill mutation for Xpath

### PR DESCRIPTION
**Pull request by Kevin222004**
Issue #13672: Kill mutation for Xpath

------------

# Mutation 

https://github.com/checkstyle/checkstyle/blob/c525c6732e2c1f15bd3f85f943770c649f1ab775/config/pitest-suppressions/pitest-xpath-suppressions.xml#L30-L46
and
https://github.com/checkstyle/checkstyle/blob/c525c6732e2c1f15bd3f85f943770c649f1ab775/config/pitest-suppressions/pitest-xpath-suppressions.xml#L3-L10

----------

# Explaination

Test added

### CODE REVIEW

File: config/pitest-suppressions/pitest-xpath-suppressions.xml [hunk 1]
```diff
@@ -27,24 +18,6 @@
...
-  </mutation>
-
-  <mutation unstable="true">
-    <sourceFile>AbstractRootNode.java</sourceFile>
-    <mutatedClass>com.puppycrawl.tools.checkstyle.xpath.AbstractRootNode</mutatedClass>
-    <mutatedMethod>iterateAxis</mutatedMethod>
-    <mutator>org.pitest.mutationtest.engine.gregor.mutators.experimental.RemoveSwitchMutator_8</mutator>
-    <description>RemoveSwitch 8 (case value 8)</description>
-    <lineContent>switch (axisNumber) {</lineContent>
-  </mutation>
```
Comment (Kevin222004) [inline config/pitest-suppressions/pitest-xpath-suppressions.xml:46]: Their is no case in which the case value 8 is used

File: config/checkstyle-checks.xml [hunk 2]
```diff
@@ -613,6 +613,11 @@
       <property name="files" value=".*[\\/]src[\\/](test|it|xdocs-examples)[\\/]"/>
       <property name="checks" value="MethodCount"/>
     </module>
+    <module name="SuppressionXpathSingleFilter">
+      <property name="files"
+                value=".*[\\/]xpath[\\/]AbstractElementNode.java"/>
+      <property name="checks" value="ImportControl"/>
```
Comment (romani) [inline config/checkstyle-checks.xml:619]: I do not see any criminal to allow new import in this class.
Please remove suppression and add allowance for required import.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: config/checkstyle-checks.xml (hunk 2 above)
Comment (Kevin222004) [reply config/checkstyle-checks.xml]: done

File: src/test/resources/com/puppycrawl/tools/checkstyle/xpath/abstractrootnode/InputAbstractRootNode.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+com.puppycrawl.tools.checkstyle.checks.coding.IllegalTokenTextCheck
+format = [^a-zA-z0-9]*
+ignoreCase = (default)false
+message = (default)
+tokens = CHAR_LITERAL
+
+*/
+package com.puppycrawl.tools.checkstyle.xpath.abstractrootnode;
+
+public class InputAbstractRootNode {
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/xpath/abstractrootnode/InputAbstractRootNode.java:15]: SuppressionXpathFilter vs InputAbstractRootNode is red flag.

please move this code to SuppressionXpathFilterTest.
fact that we have some abstract class does not mean that we need to test is as abstract. We test all from user perspective, user knows only SuppressionXpathFilter and amount of abstract classes or classes used by composite that are under it does not matter.

if you keep all tests at level of SuppressionXpathFilter we can refactor all easily and we will never need to change a test.

File: config/suppressions-xpath.xml [hunk 4]
```diff
@@ -515,4 +515,8 @@
           checks="LocalVariableNameCheck"
           query="//METHOD_DEF[./IDENT[@text='getLineNumbersFromExpected']]
                 //VARIABLE_DEF/IDENT[@text='i']"/>
+  <suppress-xpath
+          files="AbstractRootNodeTest.java"
+          checks="AbstractClassName"
```
Comment (romani) [inline config/suppressions-xpath.xml:520]: not good, see a reason in another comment.

**Review by romani — CHANGES_REQUESTED** [review]
last:

File: src/test/resources/com/puppycrawl/tools/checkstyle/xpath/abstractrootnode/InputAbstractRootNode.java (hunk 3 above)
Comment (Kevin222004) [reply src/test/resources/com/puppycrawl/tools/checkstyle/xpath/abstractrootnode/InputAbstractRootNode.java]: Done

File: config/suppressions-xpath.xml (hunk 4 above)
Comment (Kevin222004) [reply config/suppressions-xpath.xml]: Done

**Review by romani — APPROVED** [review]
Awesome

**Pull request merged**